projects/storefrontapp-e2e-cypress/cypress/integration/vendor/cds/profile-tag/default-events.e2e-spec.ts

```diff
@@ -420,6 +420,31 @@ describe('Consent Changed', () => {
   });
 });
 
+describe('Consent Changed for registered user', () => {
+  beforeEach(() => {
+    cdsHelper.setUpMocks(strategyRequestAlias);
+    navigation.visitHomePage({
+      options: {
+        onBeforeLoad: profileTagHelper.interceptProfileTagJs,
+      },
+    });
+    profileTagHelper.waitForCMSComponents();
+    anonymousConsents.clickAllowAllFromBanner();
+    loginHelper.loginAsDefaultUser();
+  });
+  it('should not send a consentgranted = false event on a page refresh', () => {
+    cy.reload();
```

Comment:
When viewing the test in a headed fashion (on chrome) ie `cypress run --config-file cypress.ci.json --spec "cypress/integration/vendor/cds/profile-tag/default-events.e2e-spec.ts" --headed`, I found that somehow the login doesn't actually go through and the check we perform immediately corresponds to an anonymous user.

This could be avoided by waiting for the loginNotification to be sent first and then reloading the page. For inspiration on how to do that, please check: https://github.com/SAP/spartacus/blob/4d17239588c73897e5f2966fd76f05c06fa02655/projects/storefrontapp-e2e-cypress/cypress/integration/vendor/cds/profile-tag/login-notification.e2e-spec.ts#L14. 

Additionally, once I reverted the fix for https://github.com/SAP/spartacus/pull/14480, one would expect this test to fail but it doesn't & hence looking at the headed version helped to identify that we expect things before a login actually happens.